js/lib/tree.js

```diff
@@ -320,6 +340,55 @@ var TreeView = widgets.DOMWidgetView.extend({
                 this.model.set('selected_nodes', selected_nodes);
                 this.model.save_changes();
             }
+        ).bind(
+            "move_node.jstree", (evt, data) => {
+
+                var new_parent = ((data.parent == "#") ? this.model : nodesRegistry[data.parent]);
+                var old_parent = ((data.old_parent == "#") ? this.model : nodesRegistry[data.old_parent]);
+
+                var np_children = [];  // new children of new_parent
+                var op_children = [];  // new children of old_parent
+
+                var update_np_quietly = false;
+                // If updating the old_parent's children changes the index of a new ancestor
```

Comment:
This is complete guess work. I don't know what is happening behind the scenes, but these are the assumptions that led to a working solution